### PR TITLE
Flake8 black

### PR DESCRIPTION
**Pull request by thejcannon**
# Justification

Now that `black` has been approved as a formatting technology, we should have `ni-python-styleguide` follow suit. This involves:

- Ensuring `ni-python-styleguide`'s code adheres to `black`'s style
- Adding a convention that all code should follow `black`'s style
- Removing any conventions which `black` automatically enforces
- Adding `flake8-black` to `ni-python-styleguide`, which enforces `black`'s style at the linter level.

### CODE REVIEW

File: ni_python_styleguide/flakehell_config.toml [hunk 1]
```diff
@@ -1,5 +1,14 @@
 # (Please list things by alphabetical order)
 
+[tool.black]
+# Projects are encouraged to configure black in their pyproject.toml, so that black can be run
+# to format the file (either directly or by editors). However, the linter will enforce that code
+# adheres to black's style with a line length of 100.
```
Comment (mshafer-NI) [inline ni_python_styleguide/flakehell_config.toml:6]: maybe add a link to https://ni.github.io/python-styleguide/#F-1-1?

File: pyproject.toml [hunk 2]
```diff
@@ -42,9 +45,13 @@ grouped = "ni_python_styleguide._vendor.flakehell.formatters:GroupedFormatter"
 json = "ni_python_styleguide._vendor.flakehell.formatters:JSONFormatter"
 stat = "ni_python_styleguide._vendor.flakehell.formatters:StatFormatter"
 
+[tool.black]
+line-length = 100
```
Comment (mshafer-NI) [inline pyproject.toml:49]: maybe add a link to https://ni.github.io/python-styleguide/#F-1-1?

File: pyproject.toml [hunk 3]
```diff
@@ -42,9 +45,13 @@ grouped = "ni_python_styleguide._vendor.flakehell.formatters:GroupedFormatter"
...
+force-exclude = '/ni_python_styleguide/_vendor|.venv/'
```
Comment (mshafer-NI) [inline pyproject.toml:50]: The default includes quite  a few more folders...
`/(\\.eggs|\\.git|\\.hg|\\.mypy_cache|\\.nox|\\.tox|\\.venv|_build|buck-out|build|dist|\\.yaml|\\.bat|\\.toml)/` I imagin .tox, .git, and config files at least should get excluded.

File: pyproject.toml [hunk 4]
```diff
@@ -42,9 +45,13 @@ grouped = "ni_python_styleguide._vendor.flakehell.formatters:GroupedFormatter"
...
+
 [tool.flakehell]
 exclude = ["ni_python_styleguide/_vendor", ".venv", "docs"]
-show_source = true
+show_source = false
```
Comment (mshafer-NI) [inline pyproject.toml:54]: why this change?

File: tests/test_convention_doc/test_codeblocks.py [hunk 5]
```diff
@@ -22,12 +24,11 @@ def test_rule_codeblocks_documents_bad_good_best(codeblock):
 def test_bad_codeblocks_document_lint_errors(lint_codeblock, bad_codeblock, capsys):
     if bad_codeblock.rule.is_automatically_enforced:
         assert not lint_codeblock(
-            bad_codeblock, "--format", "'%(code)s'"
+            bad_codeblock, "--format", "json"
```
Comment (mshafer-NI) [inline tests/test_convention_doc/test_codeblocks.py:27]: what does this cause failed blocks to render as?

**Review by irwand — APPROVED** [review]

**Review by jryckman — APPROVED** [review]

File: pyproject.toml (hunk 3 above)
Comment (thejcannon) [reply pyproject.toml:50]: We don't use any of those _in this tool_ though, so repeating them here would serve no purpose.

File: pyproject.toml (hunk 4 above)
Comment (thejcannon) [reply pyproject.toml:54]: It was making it so that the json format wasn't json-loadable.
Unfortunately there isn't a cmd-line param to turn this off, only on 😢

File: tests/test_convention_doc/test_codeblocks.py (hunk 5 above)
Comment (thejcannon) [reply tests/test_convention_doc/test_codeblocks.py:27]: A JSON parsable string (don't have an example close-by, why do you ask?)

Comment (mshafer-NI) [reply tests/test_convention_doc/test_codeblocks.py:27]: wondering if this will be easily human readable when/if the PR check fails on this.

**Review by DavidCurtiss — APPROVED** [review]
👏👏👏

**Review by rick-morrow — APPROVED** [review]

**Review by rtzoeller — APPROVED** [review]

**Review by neilvana — APPROVED** [review]

**Review by sbethur — APPROVED** [review]

File: tests/test_convention_doc/test_codeblocks.py (hunk 5 above)
Comment (thejcannon) [reply tests/test_convention_doc/test_codeblocks.py:27]: It certainly won't be illegible. I mean, it'll be json, but `{"code": "4506", "file": ...", ...}` isn't fatal.
Of course, if we want to clean up the errors, that can easily be done.

**Review by mshafer-NI — APPROVED** [review]
